lib/pipeline.js

```diff
@@ -273,7 +275,7 @@ class PipelineModel extends BaseModel {
                     return configPipeline.getConfiguration({});
                 }
 
-                return configPipeline.getConfiguration({ ref });
+                return configPipeline.getConfiguration({ ref, id: this.id });
```

Comment:
```suggestion
                return configPipeline.getConfiguration({ ref, id });
```
should use the new variable too?

Reply:
Currently that pattern probably doesn't occur, but logically it's certainly better:+1: